features/netsocket/UDPSocket.cpp

```diff
@@ -20,20 +20,32 @@
 
 UDPSocket::UDPSocket()
 {
+    _socket_stats.stats_new_socket_entry(this);
 }
 
 UDPSocket::~UDPSocket()
 {
+    _socket_stats.stats_update_socket_state(this, SOCK_CLOSED);
+}
+
+nsapi_error_t UDPSocket::close()
+{
+    _socket_stats.stats_update_socket_state(this, SOCK_CLOSED);
+    return InternetSocket::close();
 }
 
 nsapi_protocol_t UDPSocket::get_proto()
 {
+    _socket_stats.stats_update_proto(this, NSAPI_UDP);
+    _socket_stats.stats_update_socket_state(this, SOCK_OPEN);
```

Comment:
Same as in TCPSocket: type should be set in the constructor. State should be set in the `::open()` call.